File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        
        HashSet<Integer> lostSet = new HashSet<>();
        HashSet<Integer> reserveSet = new HashSet<>();
        
        for(int i=0; i<lost.length; i++){
            lostSet.add(lost[i]);
        }
        
        for(int i=0; i<reserve.length; i++){
            if(lostSet.contains(reserve[i])){
                lostSet.remove(reserve[i]);
                continue;
            }
            reserveSet.add(reserve[i]);
        }
        
        int possible = n-lostSet.size();
        
        for(int item : lostSet){
            if(reserveSet.contains(item-1)){
                possible++;
                reserveSet.remove(item-1);
            }else if(reserveSet.contains(item+1)){
                possible++;
                reserveSet.remove(item+1);
            }
        }
    
        return possible;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    
    var lostSet = new Set();
    var reserveSet = new Set();
    
    lost.sort((a,b) => a-b);
    
    for(let i=0; i<lost.length; i++){
        lostSet.add(lost[i]);
    }
    
    for(let i=0; i<reserve.length; i++){
        if(lostSet.has(reserve[i])){
            lostSet.delete(reserve[i]);
            continue;
        }
        reserveSet.add(reserve[i]);
    }
    
    var possible = n-lostSet.size;
    
    for(let item of lostSet){
        if(reserveSet.has(item-1)){
            possible++;
            reserveSet.delete(item-1);
        }else if(reserveSet.has(item+1)){
            possible++;
            reserveSet.delete(item+1);
        }
    }
    
    return possible;
}